### DISCUSSION TITLE
readContract values not being updated on real time

### DISCUSSION THREAD

**Discussion by sharozx16** [wevm/wagmi#2469] [Q&A]
I'm using readContract hook to save the values in a state and using that in a useEffect to check if the read value changes after a contract write function.

the values does not seem to update on real time, only updates when I reload the page. 

I read that these values from readContract are persisited in localstorage, how do i access and render these values from wagmi cache?

**Comment by Volchunovich** [comment]
Hello, @sharozx16 I think if you need to watch the smart-contract you should use [watchContractEvent](https://wagmi.sh/core/actions/watchContractEvent) or [watchReadContract](https://wagmi.sh/core/actions/watchReadContract).
